### PR TITLE
Contract CLI commands with Optimism

### PR DESCRIPTION
**Pull request by vezenovm**
## Related Issue Or Context

Closes: #<issue>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: cb-sol-cli/cmd/bridge.js [hunk 1]
```diff
@@ -113,6 +113,112 @@ const queryResourceId = new Command("query-resource")
...
+        log(args,`Registering generic resource ID ${args.resourceId} with contract ${args.targetContract} on handler ${args.handler}`)
+        const tx = await bridgeInstance.adminSetGenericResource(args.handler, args.resourceId, args.targetContract, args.deposit, args.execute, { gasPrice: constants.OPTIMISM_GAS_PRICE, gasLimit: constants.OPTIMISM_GAS_LIMIT})
+        await waitForTx(args.provider, tx.hash)
+    })
+
+const setBurnCmdOptimism = new Command("set-burn-optimism")
+    .description("Set a token contract as burnable in a handler")
+    .option('--bridge <address>', 'Bridge contract address', constants.BRIDGE_ADDRESS)
+    .option('--handler <address>', 'ERC20 handler contract address', constants.ERC20_HANDLER_ADDRESS)
+    .option('--tokenContract <address>', `Token contract to be registered`, constants.ERC20_ADDRESS)
```
Comment (waymobetta) [inline cb-sol-cli/cmd/bridge.js:159]: The description for `--tokenContract` should probably be consistent across all commands

File: cb-sol-cli/cmd/bridge.js [hunk 2]
```diff
@@ -113,6 +113,112 @@ const queryResourceId = new Command("query-resource")
...
+
+        const prop = await bridgeInstance.getProposal(args.chainId, args.depositNonce, args.dataHash)
+
+        console.log(prop)
+    })
+
+const queryResourceIdOptimism = new Command("query-resource-optimism")
+    .description("Query the contract address associated with a resource ID")
+    .option('--handler <address>', 'Handler contract address', constants.ERC20_HANDLER_ADDRESS)
+    .option('--resourceId <address>', `ResourceID to query`, constants.ERC20_RESOURCEID)
```
Comment (waymobetta) [inline cb-sol-cli/cmd/bridge.js:213]: Perhaps `--resourceId` description could also be consistent across commands

File: cb-sol-cli/cmd/bridge.js [hunk 3]
```diff
@@ -113,6 +113,112 @@ const queryResourceId = new Command("query-resource")
...
+    .action(async function (args) {
+        await setupParentArgs(args, args.parent.parent)
+        const bridgeInstance = new ethers.Contract(args.bridge, constants.ContractABIsOptimism.Bridge.abi, args.wallet);
+
+        const prop = await bridgeInstance.getProposal(args.chainId, args.depositNonce, args.dataHash)
+
+        console.log(prop)
+    })
+
+const queryResourceIdOptimism = new Command("query-resource-optimism")
```
Comment (waymobetta) [inline cb-sol-cli/cmd/bridge.js:210]: Semantics..

Function name `queryResourceIdOptimism` includes `Id` and does not include `Cmd` whereas other commands also relating to resource ID does not include `Id` and do include `Cmd`

`queryResourceIdOptimism` => `queryResourceCmdOptimism`

**Comment by vezenovm** [comment]
Tanya is handling optimism cli commands, keeping same commands just using --optimism flag

**Pull request closed without merging**